### ISSUE TITLE
Poor inference performance in the real robot

### ISSUE THREAD

**Issue by gzyabc** [Robbyant/lingbot-vla#28]
<img width="1191" height="777" alt="Image" src="https://github.com/user-attachments/assets/7511b039-e118-45ec-b021-7837e66b6bea" />
I have collected 200 robot trajectories and trained them using Lingbot-VLA for a total of 200,000 steps, saving checkpoints every 50,000 steps.

I tested the 100,000-step checkpoint and found the inference performance to be very poor, even though the training loss dropped to around 0.02. I am training on A100 GPUs with a batch size of 32 (16 per node). The dataset is in LeRobot format.

I am struggling to identify the root cause. Could you provide some insights? Is it more likely an issue with the dataset size, action normalization, training configuration, or deployment pipeline

**Comment by gzyabc** [comment]
@YangS03 I haven't performed an open-loop test yet. To clarify, does the procedure involve feeding three images from the original dataset into the model and comparing the predicted actions against the ground truth labels?"

I believe my dataset is solid because I previously trained a PI model (Physical Intelligence) with it, and the resulting inference actions were essentially normal. However, when using the exact same dataset with Lingbot-VLA, the predicted actions are incorrect. Here is my training configuration，could you help me check if there are any errors or misalignments?

<img width="1022" height="1170" alt="Image" src="https://github.com/user-attachments/assets/b124e071-84d8-41e2-9113-c2457f7ac4ab" />

**Comment by gzyabc** [comment]
@YangS03 Here is the info.json file. I am currently training using the CustomizedRobotwinDataset class. Given the current issues, should I be using the LiberoDataset instead? I would appreciate it if you could clarify which dataset class is best suited for this specific data format.
{
    "codebase_version": "v2.1",
    "robot_type": null,
    "total_episodes": 201,
    "total_frames": 215941,
    "total_tasks": 1,
    "total_videos": 603,
    "total_chunks": 1,
    "chunks_size": 1000,
    "fps": 30,
    "splits": {
        "train": "0:201"
    },
    "data_path": "data/chunk-{episode_chunk:03d}/episode_{episode_index:06d}.parquet",
    "video_path": "videos/chunk-{episode_chunk:03d}/{video_key}/episode_{episode_index:06d}.mp4",
    "features": {
        "observation.state": {
            "dtype": "float32",
            "shape": [
                16
            ]
        },
        "actions": {
            "dtype": "float32",
            "shape": [
                16
            ]
        },
        "observation.images.hand_left": {
            "dtype": "video",
            "shape": [
                480,
                640,
                3
            ],
            "info": {
                "video.height": 480,
                "video.width": 640,
                "video.codec": "av1",
                "video.pix_fmt": "yuv420p",
                "video.is_depth_map": false,
                "video.fps": 30,
                "video.channels": 3,
                "has_audio": false
            }
        },
        "observation.images.hand_right": {
            "dtype": "video",
            "shape": [
                480,
                640,
                3
            ],
            "info": {
                "video.height": 480,
                "video.width": 640,
                "video.codec": "av1",
                "video.pix_fmt": "yuv420p",
                "video.is_depth_map": false,
                "video.fps": 30,
                "video.channels": 3,
                "has_audio": false
            }
        },
        "observation.images.top_head": {
            "dtype": "video",
            "shape": [
                480,
                640,
                3
            ],
            "info": {
                "video.height": 480,
                "video.width": 640,
                "video.codec": "av1",
                "video.pix_fmt": "yuv420p",
                "video.is_depth_map": false,
                "video.fps": 30,
                "video.channels": 3,
                "has_audio": false
            }
        },
        "timestamp": {
            "dtype": "float32",
            "shape": [
                1
            ],
            "names": null
        },
        "frame_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "episode_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "task_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        }
    }
}

**Comment by YangS03** [comment]
@gzyabc, did you modify

```python
IMAGE_KEYS = (
    "base_0_rgb",
    "left_wrist_0_rgb",
    "right_wrist_0_rgb",
)
```

in [https://github.com/Robbyant/lingbot-vla/blob/main/lingbotvla/data/vla_data/transform.py](https://github.com/Robbyant/lingbot-vla/blob/main/lingbotvla/data/vla_data/transform.py)?

In the current version of the code, missing images are padded with zeros. Thus, none of the images in a new dataset will be used due to the key-mapping problem.
